### ISSUE TITLE
GPT Crawler cli

### ISSUE THREAD

**Issue by zerofill** [BuilderIO/gpt-crawler#168]
Anyone who wants to just enter the values via command line instead of editing the config.ts every time... I made a drop in config replacement.

So I can just do:

npm start url match maxPagesToCrawl outputFileName maxTokens

[](https://github.com/zerofill/GPT-Crawler-Config)

**Comment by balmacefa** [comment]
Yes Please!!

**Comment by zerofill** [comment]
Yeah thought I put the URL in but must not have. The one I did was in zerofill/GPT-Crawler-Config repo.
But if someone already did it, then it doesn't matter anyway lol.

**Comment by ghost** [comment]
yes

**Comment by rehan243** [comment]
Oh interesting, are you still using `config.ts` at all for defaults, or did you completely replace it with CLI args? I ask because I've seen setups where people mix both, default config for fallback values, but CLI to override when needed. Makes it a bit more flexible.

One thing to watch for: if you're parsing CLI inputs directly and not validating types (like ensuring `maxPagesToCrawl` is an integer), it might bite you later if someone passes bad data. I've had weird issues pop up from that, especially when piping inputs from scripts. 

Curious, what's your use case for `maxTokens` here? Are you trying to cap token usage per API call or across the whole crawl?
